iree/turbine/kernel/wave/codegen/handlers.py

```diff
@@ -622,6 +698,30 @@ def handle_minimum(lhs: Value, rhs: Value, options: WaveCompileOptions) -> OpRes
     return result
 
 
+@handle_atomic_op(atomic_min)
+def handle_atomic_min(
+    val: Value, buffer: Value, idx: list[Value], options: WaveCompileOptions
+) -> OpResult:
+    value_type = val.type
+    value_element_type = get_type_or_element_type(value_type)
+    atomic_kind = None
+    if _is_float_type(value_element_type):
+        atomic_kind = arith_d.AtomicRMWKind.minimumf
+    elif _is_integer_like_type(value_element_type) and _is_signed_or_signless_type(
+        value_element_type
+    ):
+        atomic_kind = arith_d.AtomicRMWKind.mins
+    num_elements = val.type.shape[0]
+    unroll_result = []
+    for i in range(num_elements):
+        a = vector_d.extract(val, static_position=[i], dynamic_position=[])
+        atomic_result = memref_d.atomic_rmw(atomic_kind, a, buffer, idx[i])
+        unroll_result.append(atomic_result)
+
+    result = vector_d.from_elements(value_type, unroll_result)
+    return result
```

Comment:
Perhaps this should be handled/moved to `handle_atomic_op` S.T we can reuse this unrolling for future atomic ops.